### PR TITLE
feat: add git hooks to project

### LINKED ISSUE #11: Set pre-commit, push hooks + python lint
Setup git hooks to pre-commit and push
Setup python linter

### PR DESCRIPTION
**Pull request by rfgvieira**
## Description

- Adds pre-push and pre-commit hooks
- Adds flake8 as python's lint
- Adds flake8 to github workflow

## Related Issues

- Closes #11

### CODE REVIEW

**Review by mari1912 — APPROVED** [review]

**Comment by lfjnascimento** [comment]
Good job 😎
I just have one question, why do we have a directory named `_` inside `husky`?

Nit: I would reverse the order of commits, so that we never have an empty `backend/pre-commit`

File: dashboard/.husky/_/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+h
```
Comment (lfjnascimento) [inline dashboard/.husky/_/.gitignore:1]: h? 🤔

Comment (rfgvieira) [reply dashboard/.husky/_/.gitignore:1]: it's the default file with configurations when you init husky, and the _ directory was created by husky

**Comment by mari1912** [comment]
The lint checking is failing for python lint, you should fix it before merge

**Pull request merged**